.circleci/config.yml

```diff
@@ -0,0 +1,253 @@
+version: 2.1
+executors:
+  default:
+    docker:
+    - image: circleci/golang:1.15.8
+
+aliases:
+- &restore_cache
+  restore_cache:
+    key: go-mod-v1-{{ checksum "go.sum" }}
+- &save_cache
+  save_cache:
+    key: go-mod-v1-{{ checksum "go.sum" }}
+    paths:
+      - "/go/pkg/mod"
+
+jobs:
+  check-style:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - *restore_cache
+    - run: make check-style
+    - *save_cache
+
+  check-modules:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - *restore_cache
+    - run: make check-modules
+    - *save_cache
+
+  check-boilerplate:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - run: ./scripts/verify-boilerplate.sh
+
+  check-mocks:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - run: make verify-mocks
+
+  test-sqlite:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - *restore_cache
+    - run: |
+        make unittest
+        make goverall
+    - *save_cache
+
+  test-postgres:
+    docker:
+    - image: circleci/golang:1.14.6
+      environment:
+        GENESIS_DATABASE=postgres://genesis_test@localhost:5432/genesis_test?sslmode=disable
+    - image: circleci/postgres:11.2-alpine
+      environment:
+        POSTGRES_USER: genesis_test
+        POSTGRES_DB: genesis_test
+    executor:
+      name: default
+    steps:
+    - run:
+        name: Waiting for Postgres to be ready
+        command: |
+          for i in `seq 1 10`;
+          do
+            nc -z localhost 5432 && echo Success && exit 0
+            echo -n .
+            sleep 1
+          done
+          echo Failed waiting for Postgres && exit 1
+    - checkout
+    - *restore_cache
+    - run: make unittest
+    - *save_cache
+
+  push-docker-pr:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - setup_remote_docker
+    - run: make build-image
+    - run: ./scripts/push-docker-pr.sh
+    - run:
+        name: Re-run Docker Push if fail
+        command: |
+            sleep 20
+            ./scripts/push-docker-pr.sh
+        when: on_fail
+
+  push-docker:
+    executor:
+      name: default
+    steps:
+    - checkout
+    - setup_remote_docker
+    - run: make build-image
+    - run: ./scripts/push-docker.sh
+    - run:
+        name: Re-run Docker Push if fail
+        command: |
+            sleep 20
+            ./scripts/push-docker.sh
+        when: on_fail
+
+  scan-image:
+    docker:
+    - image: registry.gitlab.com/gitlab-org/security-products/analyzers/klar:latest
+      environment:
+        GIT_STRATEGY: none
+        CI_APPLICATION_REPOSITORY: mattermost/genesis
+        CLAIR_DB_CONNECTION_STRING: "postgresql://postgres:password@localhost:5432/postgres?sslmode=disable&statement_timeout=60000"
+        DOCKERFILE_PATH: "build/Dockerfile"
+    - image: arminc/clair-db:latest
+    steps:
+    - checkout
+    - run: |
+        export CI_APPLICATION_TAG="${CIRCLE_SHA1:0:7}"
```

Comment:
I think you can do this as part of the run 
```
environment:
 CI_APPLICATION_TAG="${CIRCLE_SHA1:0:7}" <--- not sure if this works though, eg. in github actions doesn't if I recall
 DOCKER_USER=$DOCKER_USERNAME
```

Reply:
Yeap nice, changed it :)